helios-client/src/main/java/com/spotify/helios/common/descriptors/Job.java

```diff
@@ -212,6 +223,8 @@ public Job(@JsonProperty("id") final JobId id,
     this.securityOpt = Optional.fromNullable(securityOpt).or(EMPTY_SECURITY_OPT);
     this.networkMode = Optional.fromNullable(networkMode).orNull();
     this.metadata = Optional.fromNullable(metadata).or(EMPTY_METADATA);
+    this.capAdds = Optional.fromNullable(capAdds).or(EMPTY_CAPS);
+    this.capDrops = Optional.fromNullable(capDrops).or(EMPTY_CAPS);
```

Comment:
pretty minor, and this pattern exists already, but even the guava Optional docs recommend using something like `Objects.firstNonNull(a, b)` for this use case: https://github.com/google/guava/wiki/UsingAndAvoidingNullExplained#convenience-methods